async.js

```diff
@@ -1,50 +1,152 @@
 // Task #1
 //
-// Write sumArray function, that will accept array, success callback, failure callback
-// and return nothing. It should summarize all elements and pass sum in the success cb.
+// Write sumArray function, that will accept array, success callback,
+// failure callback and return nothing. It should summarize all elements
+// and pass sum in the success cb.
 // if error occurs or elements are not numbers - it should invoke fail cb.
-//
-// sumArray([1, 2, 3, 4, 5], console.log, console.log)
-// 15
-//
-// sumArray([1, 2, {}, 4, 5], console.log, ()=> console.log('error'))
-// error
+
+function sumArray(arr, sucCb, failCb) {
+  const sum = arr.reduce((acc, val) => acc + val, 0);
+  if (Number(sum)) sucCb(sum);
+  if (!Number(sum)) failCb(sum);
+}
+sumArray([1, 2, 3, 4, 5], console.log, console.log);
+sumArray([1, 2, {}, 4, 5], console.log, () => console.log("error"));
 
 // Task #2
 
-// Write a function that will make GET request to 'https://api.kanye.rest' - if it has a quote, make another one. No async/await
+// Write a function that will make GET request
+// to 'https://api.kanye.rest' - if it has a quote,
+// make another one. No async/await
+
+function getData() {
+  return fetch("https://api.kanye.rest")
+    .then((responce) => responce.json())
+    .then((data) => console.log(data));
+}
+
+getData().then(getData());
 
 // Task #3
 
-// Write a function that will make 5 sequential and 5 parallel requests to 'https://api.kanye.rest';
+// Write a function that will make 5 sequential
+// and 5 parallel requests to 'https://api.kanye.rest';
+
+function getDataFromKanye() {
+  // 5 parallel requests
+  const urls = [
+    "https://api.kanye.rest",
+    "https://api.kanye.rest",
+    "https://api.kanye.rest",
+    "https://api.kanye.rest",
+    "https://api.kanye.rest",
+  ];
+  Promise.all(
+    urls.map((url) =>
+      fetch(url)
+        .then((responce) => responce.json())
+        .then((data) => console.log(data))
+    )
+  );
+
+  // 5 sequential requests
+
+  const sequental = async function () {
+    const responce = await fetch("https://api.kanye.rest");
+    const data = await responce.json();
+    console.log(data);
+  };
+
+  return sequental()
```

Comment:
Even though this is totally correct answer, the ideal logic would be to implement function that will do something declarative and similar to Promise.all.